File: endpoints/photos.py
from silence.decorators import endpoint

@endpoint(
    route="/photos",
    method="GET",
    sql="SELECT * FROM Photos WHERE visibility = 'Public' ORDER BY DATE DESC "
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/photos/user/$userId",
    method="GET",
    sql="SELECT * FROM Photos WHERE (userId != $userId && visibility = 'Public') || (userId = $userId) ORDER BY date DESC"
)
def get_all_reverse():
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="GET",
    sql="SELECT * FROM Photos WHERE photoId = $photoId "
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/categories/$categoryId",
    method="GET",
    sql="SELECT * FROM Photos WHERE categoryId = $categoryId && visibility = 'Public'"
)
def get_by_category():
    pass

###############################################################################

@endpoint(
    route="/photos",
    method="POST",
    sql="INSERT INTO Photos (title, description, url, visibility, userId, categoryId) VALUES ($title, $description, $url, $visibility, $userId, $categoryId)",
    description="Creates a new photo",
    auth_required=True,
)
def create(title, description, url, visibility, userId, categoryId):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="PUT",
    sql="UPDATE Photos SET title = $title, description = $description, url = $url, visibility = $visibility, rate = $rate , numRaters = $numRaters, categoryId = $categoryId WHERE photoId = $photoId",
    description="Updates an existing photo",
    auth_required=True,
)
def update(title, description, url, visibility, rate, numRaters, categoryId):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="DELETE",
    sql="DELETE FROM Photos WHERE photoId = $photoId",
    description="Removes a photo",
    auth_required=True,
)
def delete():
    pass

File: docs/js/api/photos.js
"use strict";
import { BASE_URL, requestOptions } from "./common.js";

const photosAPI = {
    getAll: function () {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getAllReverse: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/user/${userId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getAllReversebyUserId: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/users/photos/${userId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getAllPrivatebyUserId: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/user/photos/${userId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getById: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getByCategoryId: function (categoryId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/categories/${categoryId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    create: function (formData) {
        return new Promise(function (resolve, reject) {
            axios
                .post(`${BASE_URL}/photos`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    update: function (photoId, formData) {
        return new Promise(function (resolve, reject) {
            axios
                .put(`${BASE_URL}/photos/${photoId}`, formData,
                    requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    delete: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .delete(`${BASE_URL}/photos/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    }


};

export { photosAPI };